packages/iterator-utilities/README.md

```diff
@@ -0,0 +1,765 @@
+<div align="center">
+
+![Sapphire Logo](https://raw.githubusercontent.com/sapphiredev/assets/main/banners/SapphireCommunity.png)
+
+# @sapphire/iterator-utilities
+
+**Iterator utilities for JavaScript.**
+
+[![GitHub](https://img.shields.io/github/license/sapphiredev/utilities)](https://github.com/sapphiredev/utilities/blob/main/LICENSE.md)
+[![codecov](https://codecov.io/gh/sapphiredev/utilities/branch/main/graph/badge.svg?token=OEGIV6RFDO)](https://codecov.io/gh/sapphiredev/utilities)
+[![npm bundle size](https://img.shields.io/bundlephobia/min/@sapphire/iterator-utilities?logo=webpack&style=flat-square)](https://bundlephobia.com/result?p=@sapphire/iterator-utilities)
+[![npm](https://img.shields.io/npm/v/@sapphire/iterator-utilities?color=crimson&logo=npm&style=flat-square)](https://www.npmjs.com/package/@sapphire/iterator-utilities)
+
+</div>
+
+**Table of Contents**
+
+-   [Features](#features)
+-   [Installation](#installation)
+-   [Usage](#usage)
+    -   [Resolving](#resolving)
+    -   [Checking for the existence of any bit](#checking-for-the-existence-of-any-bit)
+    -   [Checking for the existence of one or multiple bits](#checking-for-the-existence-of-one-or-multiple-bits)
+    -   [Getting the field's complement](#getting-the-fields-complement)
+    -   [Getting the union of different fields](#getting-the-union-of-different-fields)
+    -   [Getting the intersection between different fields](#getting-the-intersection-between-different-fields)
+    -   [Getting the symmetric difference between different fields](#getting-the-symmetric-difference-between-different-fields)
+    -   [Formatting a field](#formatting-a-field)
+-   [Buy us some doughnuts](#buy-us-some-doughnuts)
+-   [Contributors ✨](#contributors-)
+
+## Features
+
+-   Written in TypeScript
+-   Bundled with esbuild so it can be used in NodeJS and browsers
+-   Offers CommonJS, ESM and UMD bundles
+-   Fully tested
+
+## Installation
+
+You can use the following command to install this package, or replace `npm install` with your package manager of choice.
+
+```sh
+npm install @sapphire/iterator-utilities
+```
+
+## Usage
+
+For any of the following examples, you can import the utilities from the index file:
+
+```ts
+import { append } from '@sapphire/iterator-utilities';
+const { append } = require('@sapphire/iterator-utilities');
+```
+
+Or you can import the utilities directly:
+
+```ts
+import { append } from '@sapphire/iterator-utilities/append';
+const { append } = require('@sapphire/iterator-utilities/append');
```

Comment:
Does Require syntax even support that? Pretty sure it doesn't...?

Reply:
Yes, it does, paths aren't an ESM-only feature after all

Reply:
Yes, it does. In my screenshot it _would've_ worked if i wasn't in highlight where its all esm

![iTerm2 - 2024-04-28 at 15 28 53](https://github.com/sapphiredev/utilities/assets/17960496/993b96fe-2207-48a3-9d70-39e9e1c7a811)